src/cmd/ship.go

```diff
@@ -106,12 +108,18 @@ func getShipStepList(config shipConfig) (result steps.StepList) {
 	result.Append(&steps.MergeBranchStep{BranchName: branchToMergeInto})
 	result.Append(&steps.EnsureHasShippableChangesStep{BranchName: config.BranchToShip})
 	result.Append(&steps.CheckoutBranchStep{BranchName: branchToMergeInto})
-	result.Append(&steps.SquashMergeBranchStep{BranchName: config.BranchToShip, CommitMessage: commitMessage})
+	canUseDriver := canUseDriver(config.BranchToShip, branchToMergeInto)
+	if canUseDriver {
+		result.Append(&steps.DriverMergePullRequestStep{BranchName: config.BranchToShip, CommitMessage: commitMessage})
+		result.Append(&steps.PullBranchStep{})
+	} else {
+		result.Append(&steps.SquashMergeBranchStep{BranchName: config.BranchToShip, CommitMessage: commitMessage})
+	}
 	if git.HasRemote("origin") {
 		result.Append(&steps.PushBranchStep{BranchName: branchToMergeInto, Undoable: true})
 	}
 	childBranches := git.GetChildBranches(config.BranchToShip)
-	if git.HasTrackingBranch(config.BranchToShip) && len(childBranches) == 0 {
+	if canUseDriver || (git.HasTrackingBranch(config.BranchToShip) && len(childBranches) == 0) {
```

Comment:
If we merge with a driver then we can always delete the remote branch. If we aren't using a driver, then we can't delete the remote branch if this branch has children